phpBB/includes/mcp/mcp_warn.php

```diff
@@ -530,12 +530,30 @@ function add_warning($user_row, $warning, $send_pm = true, $post_id = 0)
 		include_once($phpbb_root_path . 'includes/functions_privmsgs.' . $phpEx);
 		include_once($phpbb_root_path . 'includes/message_parser.' . $phpEx);
 
-		$user_row['user_lang'] = (file_exists($phpbb_root_path . 'language/' . $user_row['user_lang'] . "/mcp.$phpEx")) ? $user_row['user_lang'] : $config['default_lang'];
-		include($phpbb_root_path . 'language/' . basename($user_row['user_lang']) . "/mcp.$phpEx");
+		// Okay we need to get the reason from the warned users language
+		// Only load up the language pack if the language is different to the current one
+		if ($user_row['user_lang'] != $user->lang_name)
+		{
+			// Load up the language pack
+			$lang = array();
+
+			$user_row['user_lang'] = (file_exists($phpbb_root_path . 'language/' . $user_row['user_lang'] . "/mcp.$phpEx")) ? $user_row['user_lang'] : $config['default_lang'];
+			include($phpbb_root_path . 'language/' . basename($user_row['user_lang']) . "/mcp.$phpEx");
```

Comment:
> use  $user_row['user_lang']  in both cases

The same code is [here](https://github.com/phpbb/phpbb/blob/3.1.x/phpBB/includes/mcp/mcp_queue.php#L1253).
I treated it as an example.